### DISCUSSION TITLE
Sorting/Filtering a table

### DISCUSSION THREAD

**Discussion by avrame** [remix-run/remix#5935] [Q&A] [answered]
Hi all,
I'm not sure which approach I should use to sort/filter a table. This is a paginated table of data that I'm fetching from a Strapi (headless CMS) server. I'd like the user to be able to click on the sort icon in the table column header and have it toggle between ascending and descending. Currently, the page loads the table data using the exported `loader` function, but I'm not sure if I can re-use that for sorting and filtering - maybe using URL query params?
Thanks for any advice,
Avi

**Comment by kiliman** [comment, marked as answer]
I have an example repo that shows sorting, filtering, and paging.

https://github.com/kiliman/remix-scoped-params

You can see it in action here: https://remix-scoped-params-production.up.railway.app/

**Reply by avrame** [reply]
Thanks, that's exactly what I need!

**Reply by kiliman** [reply]
FYI, scopes are optional. They're there to allow you to maintain a different set of search params across different URLs. In this case, across tabs.

The main benefit of the helper is that it maintains the existing search params. You're simply providing an updated value. So for sorting or paging, it returns an URL where the query string only updates that specific value and the rest will remain as the current URL.

**Reply by avrame** [reply]
I ended up using `useSearchParams()` and it works great. I used your `URL.searchParams.get()` method too.

**Reply by kiliman** [reply]
Awesome. As long as you got the idea, I'm glad you got something working.
